db/snapshot-ctl.cc

```diff
@@ -209,13 +199,13 @@ future<int64_t> snapshot_ctl::true_snapshots_size() {
     return run_snapshot_list_operation([this] {
         return _db.map_reduce(adder<int64_t>(), [] (database& db) {
             return do_with(int64_t(0), [&db] (auto& local_total) {
-                return parallel_for_each(db.get_column_families(), [&local_total] (auto& cf_pair) {
-                    return cf_pair.second->get_snapshot_details().then([&local_total] (auto map) {
-                        for (auto&& snap_map: map) {
-                            local_total += snap_map.second.live;
+                return db.get_snapshot_details().then([&local_total] (auto&& map) {
```

Comment:
Is it possible for the map to be very big and cause the following nested loops to create a reactor stall? If so, maybe it would make sense to use one of seastar loops and check need_preempt() inside?

Reply:
I don't think it can, the size of the map is equal to the number of snapshots - which shouldn't be that big?

Reply:
Those are snapshots for all tables, no? Meaning that there can be thousands of them.

Reply:
Well yeah, but is thousands a very big number tho?